File: frontend/src/api/brand.js
// 品牌分析相关 API, 全部走 /api 代理(5173 -> 8080)
// 注意: 响应拦截器返回的是完整 { code, data } 信封, 这里统一拆出 data,
// 使 store 拿到的就是业务数组/对象(与 store 内部的数组校验一致)。
import request from './request'

const brand = {
  kpi:       (q) => request.get('/brand/kpi', { params: q }).then(r => r.data),
  trend:     (q) => request.get('/brand/trend', { params: q }).then(r => r.data),
  platforms: (q) => request.get('/brand/platforms', { params: q }).then(r => r.data),
  sentiment: (q) => request.get('/brand/sentiment', { params: q }).then(r => r.data),
  keywords:  (q) => request.get('/brand/keywords', { params: q }).then(r => r.data),
  list:      (q) => request.get('/brand/list', { params: q }).then(r => r.data)
}

export default brand

File: backend/internal/api/handler/brand.go
package handler

import (
	"net/http"

	"github.com/gin-gonic/gin"

	"insta360-insight/internal/model"
	"insta360-insight/internal/service"
)

type Brand struct{ svc *service.BrandService }

func NewBrand(svc *service.BrandService) *Brand { return &Brand{svc: svc} }

func (h *Brand) Kpi(c *gin.Context) {
	d, e := h.svc.Kpi(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Brand) Trend(c *gin.Context) {
	d, e := h.svc.Trend(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Brand) Platforms(c *gin.Context) {
	d, e := h.svc.Platforms(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Brand) Sentiment(c *gin.Context) {
	d, e := h.svc.Sentiment(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Brand) Keywords(c *gin.Context) {
	d, e := h.svc.Keywords(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
func (h *Brand) List(c *gin.Context) {
	d, e := h.svc.List(c.Request.Context(), bindFilter(c))
	if e != nil {
		fail(c, e)
		return
	}
	c.JSON(http.StatusOK, model.OK(d))
}
